### PR TITLE
delay support

### PR DESCRIPTION
**Pull request by adammoisa**

### CODE REVIEW

File: src/FadeIn.js [hunk 1]
```diff
@@ -7,25 +7,28 @@ export default class FadeIn extends Component {
 
   componentDidMount() {
     const count = React.Children.count(this.props.children);
+    var customLag = parseInt(this.props.lag) ? parseInt(this.props.lag) : 50;
```
Comment (gkaemmer) [inline src/FadeIn.js:10]: Any reason to `parseInt` here? Might make the API simpler to just accept a number of milliseconds.

File: src/FadeIn.js [hunk 2]
```diff
@@ -7,25 +7,28 @@ export default class FadeIn extends Component {
...
     let i = 0;
     this.interval = setInterval(() => {
       i++;
       if (i > count)
         clearInterval(this.interval);
 
       this.setState({ maxIsVisible: i });
-    }, 50);
+    }, customLag);
```
Comment (gkaemmer) [inline src/FadeIn.js:18]: Just pass the prop directly here: `this.props.lag`.

Comment (gkaemmer) [reply src/FadeIn.js:18]: Or maybe call the prop `delayStart`, so you have `delay` and `delayStart`. My guess is that `delay` is the more useful prop, but I'm not sure.

File: src/FadeIn.js [hunk 3]
```diff
@@ -7,25 +7,28 @@ export default class FadeIn extends Component {
...
   }
 
   render() {
     const duration = 0.4;
+    const delay = this.props.delay;
```
Comment (gkaemmer) [inline src/FadeIn.js:23]: `const delay = this.props.delay || 0;`

File: src/FadeIn.js [hunk 4]
```diff
@@ -7,25 +7,28 @@ export default class FadeIn extends Component {
...
     return (
       <div>
         {React.Children.map(this.props.children, (child, i) => {
           return (
             <div
               style={{
                 transition: `opacity ${duration}s, top ${duration}s`,
+                'transition-delay': delay,
```
Comment (gkaemmer) [inline src/FadeIn.js:31]: I'd CSS-ify the value for the user here:
```
transitionDelay: `${delay}ms`
```

**Review by gkaemmer — CHANGES_REQUESTED** [review]
Thanks for this! Just recommended a couple changes.

**Comment by bcanseco** [comment]
Just came across this PR - it'd be super cool to have this functionality in a new version of react-fade-in. @adammoisa any chance you're still looking to get this merged?

**Comment by gkaemmer** [comment]
@bcanseco I went ahead and added `delay` and `transitionDuration` props: https://github.com/gkaemmer/react-fade-in/releases/tag/v0.1.6

**Pull request closed without merging**